### ISSUE TITLE
julia & freebsd

### ISSUE THREAD

**Issue by devosalain** [JuliaLang/julia#53604]
Is Julia directed towards linux ?
Many Julia packages don't work under freebsd but they do under artix-linux.
Examples : "plotly package, plot package, dataframe package". Important packages.
Where should i post detailed problem reports? 
On the FreeBSD bugforum or here ?

**Comment by KristofferC** [comment]
I'm surprisded DataFrames would not work on FreeBSD since it is pure Julia. What error do you have with it?

**Comment by devosalain** [comment]
```
using Pkg
Pkg.add("DataFrames")
```
Returns
```
Resolving package versions...
  No Changes to `~/.julia/environments/v1.9/Project.toml`
  No Changes to `~/.julia/environments/v1.9/Manifest.toml`
  0 dependencies successfully precompiled in 23 seconds. 251 already precompiled. 43 skipped during auto due to previous errors.
  1 dependency errored. To see a full report either run `import Pkg; Pkg.precompile()` or load the package
```
Pkg.precompile()
```
ERROR: The following 4 direct dependencies failed to precompile:
FFMPEG [c87230d0-a227-11e9-1b43-d7ebe4e7570a]
Failed to precompile FFMPEG [c87230d0-a227-11e9-1b43-d7ebe4e7570a] to "/home/x/.julia/compiled/v1.9/FFMPEG/jl_uriSZj".
ERROR: LoadError: InitError: could not load library "/home/x/.julia/artifacts/ce9ecc22b48f5833c1b89a75263d1d57ab64585a/lib/libfreetype.so"
Shared object "libz.so.1" not found, required by "libfreetype.so"
ERROR: LoadError: Failed to precompile FFMPEG_jll
ERROR: LoadError: InitError: could not load library "/home/x/.julia/artifacts/ce9ecc22b48f5833c1b89a75263d1d57ab64585a/lib/libfreetype.so"
Shared object "libz.so.1" not found, required by "libfreetype.so"
ERROR: LoadError: Failed to precompile libass_jll
ERROR: LoadError: Failed to precompile FFMPEG_jll
ERROR: LoadError: Failed to precompile FFMPEG
ERROR: LoadError: Cannot locate artifact 'electronjs_app' for x86_64-unknown-freebsd-libgfortran5-cxx11-julia_version+1.9.4
ERROR: LoadError: Failed to precompile Blink
```
I have '/usr/lib/libz.so"

**Comment by giordano** [comment]
[`FreeType2_jll.jl`](https://github.com/JuliaBinaryWrappers/FreeType2_jll.jl) depends on [`Zlib_jll.jl`](https://github.com/JuliaBinaryWrappers/Zlib_jll.jl), which [on FreeBSD provides `libz.so.1`](https://github.com/JuliaBinaryWrappers/Zlib_jll.jl/blob/2da19e459974a587ad532596b6ca1416ba37ae18/src/wrappers/x86_64-unknown-freebsd.jl#L5).  @ararslan any clue of what's going on?

**Comment by devosalain** [comment]
I use FreeBSD-14.0-Release-p3.
Julia works fine, but Julia packages fails. It's like it can't find the good dependend libraries on FreeBSD.
On Artix-linux everything works fine.

**Comment by giordano** [comment]
@devosalain  How did you install Julia?  This reminded me of [this thread on zulip](https://julialang.zulipchat.com/#narrow/stream/274208-helpdesk-.28published.29/topic/.E2.9C.94.20FreeBSD.20libz.2Eso.2E1.20not.20found/near/324856210), where it turned out the user installed Julia via ports, instead of using the [official binaries](https://julialang.org/downloads/).

**Comment by KristofferC** [comment]
>```
>ERROR: The following 4 direct dependencies failed to precompile:
>FFMPEG [c87230d0-a227-11e9-1b43-d7ebe4e7570a]
>```

DataFrames does not depend on FFMPEG, you probably have a big environment with a bunch of other packages.

**Comment by devosalain** [comment]
curl -fsSL https://install.julialang.org | sh
returns :
info: downloading installer
curl: (22) The requested URL returned error: 404
juliaup: installer for platform 'x86_64-unknown-freebsd' not found, this may be unsupported

**Comment by KristofferC** [comment]
Try downloading from https://julialang.org/downloads/#current_stable_release

**Comment by giordano** [comment]
@devosalain you didn't answer the question of how you installed Julia. Is your message meant to be a confirmation you used ports?

**Comment by ararslan** [comment]
> Is Julia directed towards linux ?

Nope! Julia officially supports Linux, macOS, Windows, and FreeBSD.

> any clue of what's going on?

@giordano: Based on the discussion here and the fact that 1.9 is being used, it sounds like a Ports build to me. (Julia 1.9.4 is the latest available version in FreeBSD's package manager and 1.9 is in the environment paths printed when installing DataFrames.)

> juliaup: installer for platform 'x86_64-unknown-freebsd' not found, this may be unsupported

@devosalain: As you observed, FreeBSD is not currently supported by the installer/version multiplexer; you'll need to download and install binaries manually from the page Kristoffer linked. That should fix your issue. If you're interested in FreeBSD support for JuliaUp, I encourage you to open an issue on (or contribute to!) https://github.com/JuliaLang/juliaup.

**Comment by devosalain** [comment]
Currently using julia version 1.10.2. From the julia website.
You just can't close this issue.
Julia&FreeBSD&Dataframes don't work. Why no idea ?
I receive the error "Error: Missing source file for DataFrames".

**Comment by KristofferC** [comment]
This issue is not locked for discussion but so far there hasn't been anything that shows that the official Julia is not working correctly on FreeBSD.

Can you try "start over", that is delete `.julia/packages`, create a new environment, install DataFrames and see what happens?

**Comment by ararslan** [comment]
> Indeed after doing this the problem "dataframes" went away.

Glad to hear it!

> I still cannot install plolty, but i must check my "electron library" install

Can you post the full output of running `using Pkg; Pkg.install("Plotly")`?

**Comment by oikumene** [comment]
JFYI. The original problem was with Julia from FreeBSD's ports&packages (explained in https://forums.freebsd.org/threads/plot-libraries-for-julia-error.92591/ ). This problem was resolved along with updating the port to 1.10.3. The package does not exist yet but will be built in a few days. (I know officially built Julia binary is recommended.)
The problem with plotly is that it requires electron app which does not support FreeBSD officially. I did not check myself but there should not exist the electron app artifact for FreeBSD. However, the electron app exists in FreeBSD's ports&packages, so there is a way to build it. I do not know how the artifact is built, nor did I prepared the electron app port for FreeBSD, but I wish I could be of some help for building it.
